### PR TITLE
Feat: add useGetRecordId documentation

### PR DESCRIPTION
**Pull request by arimet**

### CODE REVIEW

File: docs/useGetRecordId.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+layout: default
+title: "useGetRecordId"
+---
+
+# `useGetRecordId`
+
+This hook returns the current `recordId`. `recordId` is obtained from parameters if passed as a parameter, or from the `RecordContext` if there is one, or, lastly, from the react-router URL.
+
+`useGetRecordId` takes optional argument `recordId` if used inside a RecordContextProvider or if recordId can be guessed from the URL
```
Comment (djhi) [inline docs/useGetRecordId.md:10]:
```suggestion
`useGetRecordId` takes an optional `recordId` argument if used inside a `RecordContextProvider` or if `recordId` can be guessed from the URL
```

**Review by djhi — CHANGES_REQUESTED** [review]

File: docs/useGetRecordId.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+layout: default
+title: "useGetRecordId"
+---
+
+# `useGetRecordId`
+
+This hook returns the current `recordId`. `recordId` is obtained from parameters if passed as a parameter, or from the `RecordContext` if there is one, or, lastly, from the react-router URL.
+
+`useGetRecordId` takes an optional `recordId` argument if used inside a `RecordContextProvider` or if `recordId` can be guessed from the URL
```
Comment (slax57) [inline docs/useGetRecordId.md:10]: I'm not fond of the wording, which suggests the `recordId` argument is needed if used inside a `RecordContextProvider` or if `recordId` can be guessed from the URL, when it is actually the opposite.

File: docs/useGetRecordId.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+```jsx
+import { useListContext, useUnselect } from 'react-admin';
```
Comment (slax57) [inline docs/useGetRecordId.md:13]:
```suggestion
import { useGetRecordId } from 'react-admin';
```

**Review by slax57 — CHANGES_REQUESTED** [review]

**Review by slax57 — APPROVED** [review]

File: docs/navigation.html [hunk 4]
```diff
@@ -33,6 +33,7 @@
   <li {% if page.path == 'useDeleteMany.md' %} class="active" {% endif %}><a class="nav-link" href="./useDeleteMany.html"><code>useDeleteMany</code></a></li>
   <li><a class="nav-link external" href="https://marmelab.com/ra-enterprise/modules/ra-realtime#real-time-hooks.html" target="_blank"><code>useSubscribe</code><img class="premium" src="./img/premium.svg" /></a></li>
   <li {% if page.path == 'useGetTree.md' %} class="active" {% endif %}><a class="nav-link" href="./useGetTree.html"><code>useGetTree</code><img class="premium" src="./img/premium.svg" /></a></li>
+  <li {% if page.path == 'useGetRecordId.md' %} class="active" {% endif %}><a class="nav-link" href="./useGetRecordId.html"><code>useGetRecordId</code></a></li>
```
Comment (fzaninotto) [inline docs/navigation.html:36]: It shouldn't be in this section, but rather in the Hooks section with useRecordContext

**Review by fzaninotto — CHANGES_REQUESTED** [review]

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**